x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/engine_analytics_lens_logic.ts

```diff
@@ -0,0 +1,135 @@
+/*
```

Comment:
I might be missing something here, but I wonder whether we really need this lens logic file and can't just do it all in the component itself. As far as I can tell, this logic file is mostly just pushing data around--we're not making any API calls, we're not really deriving any state. We're just making things hard on ourselves by forcing all of the data through Kea, without really using that. 

Kea's at its best when we can use it to a) share logic across components and b) we have some asynchronous state that or may not be populated depending on API calls, and we need an abstraction layer to have our components deal with that nicely.

As far as I can tell, we're only using this logic in one component--so we're not using A). And all of the logic we've pushed into this is just "set static data, get static data", which suggests we're not really abstracting away much either. 

Am I missing something here?

Reply:
Based on with @joemcelroy. I'll probably drop the changes with Kea. 
Move all the states change from  [engine_analytics_logic.ts](https://github.com/elastic/kibana/blob/4c42ce0624c1ca95126b04a75786f2e7f9a24a20/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics_logic.ts) to [engines_analytics.tsx](https://github.com/elastic/kibana/blob/4c42ce0624c1ca95126b04a75786f2e7f9a24a20/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics.tsx)

Drop [ lens_logic.ts](https://github.com/elastic/kibana/blob/4c42ce0624c1ca95126b04a75786f2e7f9a24a20/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/engine_analytics_lens_logic.ts) and fetch lens data directly from` engines_analytics.tsx`

Reply:
@joemcelroy  As we agreed on removing all Kea logic from overview page. I removed lens_logic file but thought of keeping analytics logic [file](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics_logic.ts).  
Reason: While implementing, I realized not having a kea state would cause more trouble as we would some values to be shared across component.

 For example: Total Count and Percentage calculated from lens [component ](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/engine_analytics_lens.tsx#L353) is consumed by analytics_card[ component](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics_card.tsx)
Moreover, figured that having a logic file can also help us with loading default data view. What do you think?